benchmarks/nel/project.yml

```diff
@@ -36,96 +47,94 @@ workflows:
+    outputs:
+      - assets/mewsli_9
 
   - name: preprocess
-    help: Preprocess test datasets
+    help: Preprocess test datasets.
     script:
-      - "unzip -q -d assets/reddit assets/reddit.zip"
-      - "python ./scripts/clean_data.py reddit"
+      - "python ./scripts/clean_data.py ${vars.dataset}"
     deps:
-      - "assets/reddit.zip"
+      - "assets/${vars.dataset}"
     outputs:
-      - "assets/reddit"
+      - "assets/${vars.dataset}"
 
   - name: download_model
-    help: "Download a model with pretrained vectors and NER component"
+    help: "Download a model with pretrained vectors and NER component."
     script:
       - "python -m spacy download ${vars.vectors_model}"
 
   - name: parse_wiki_dumps
-    help: "Parses Wikipedia dumps."
+    help: "Parse Wikipedia dumps. This can take a long time!"
     script:
-      - "env PYTHONPATH=scripts python ./scripts/parse_wiki_dumps.py"
+      - "env PYTHONPATH=scripts python ./scripts/parse_wiki_dumps.py ${vars.use_filtered_dumps}"
+    outputs:
+      - "assets/wiki/wiki.sqlite3"
 
   - name: create_kb
-    help: "Create the knowledge base and write it to file"
+    help: "Create the knowledge base and write it to file."
     script:
-      - "python ./scripts/create_kb.py reddit ${vars.vectors_model}"
+      - "python ./scripts/create_kb.py ${vars.dataset} ${vars.vectors_model}"
     deps:
-      - "assets/reddit"
+      - "assets/${vars.dataset}"
+      - "assets/wiki/wiki.sqlite3"
     outputs:
-      - "assets/reddit/entities.pkl"
-      - "assets/reddit/entities_failed_lookup.pkl"
-      - "assets/reddit/annotations.pkl"
+      - "assets/${vars.dataset}/entities.pkl"
+      - "assets/${vars.dataset}/entities_failed_lookup.pkl"
+      - "assets/${vars.dataset}/annotations.pkl"
     outputs_no_cache:
-      - "temp/reddit/kb"
-      - "temp/reddit/nlp"
+      - "temp/${vars.dataset}/kb"
+      - "temp/${vars.dataset}/nlp"
 
   - name: compile_corpora
-    help: "Compile corpora, separated in in train/dev/test sets"
+    help: "Compile corpora, separated in in train/dev/test sets."
     script:
-      - "python ./scripts/compile_corpora.py reddit"
+      - "python ./scripts/compile_corpora.py ${vars.dataset} True"
     deps:
-      - "assets/reddit/entities.pkl"
-      - "assets/reddit/entities_failed_lookups.pkl"
-      - "assets/reddit/annotations.pkl"
-      - "temp/reddit/kb"
-      - "temp/reddit/nlp"
+      - "assets/${vars.dataset}/entities.pkl"
+      - "assets/${vars.dataset}/entities_failed_lookups.pkl"
+      - "assets/${vars.dataset}/annotations.pkl"
+      - "temp/${vars.dataset}/kb"
+      - "temp/${vars.dataset}/nlp"
       - "configs/datasets.yml"
     outputs_no_cache:
-      - "corpora/reddit/train.spacy"
-      - "corpora/reddit/dev.spacy"
-      - "corpora/reddit/test.spacy"
+      - "corpora/${vars.dataset}/train.spacy"
+      - "corpora/${vars.dataset}/dev.spacy"
+      - "corpora/${vars.dataset}/test.spacy"
 
   - name: train
```

Comment:
I do get an error about data not being properly formatted, is this a worry? It does seem to start training afterwards so I'm not sure when exactly that warning is thrown or whether it's significant.

```
E    #       LOSS ENTIT...  SENTS_F  SENTS_P  SENTS_R  ENTS_F  ENTS_P  ENTS_R  NEL_MICRO_F  NEL_MICRO_R  NEL_MICRO_P  SCORE
---  ------  -------------  -------  -------  -------  ------  ------  ------  -----------  -----------  -----------  ------
...\spacy\pipeline\legacy\entity_linker.py:193: UserWarning: [W093] Could not find any data to train the Entity Linker on. Is your input data correctly formatted?
  warnings.warn(Warnings.W093.format(name="Entity Linker"))
  0       0           0.00    88.50    91.58    85.62    0.00    0.00    0.00        98.46        98.46        98.46    0.62
  0     200          23.64    88.50    91.58    85.62    0.00    0.00    0.00        98.46        98.46        98.46    0.62
  0     400          11.84    88.50    91.58    85.62    0.00    0.00    0.00        98.46        98.46        98.46    0.62
```

Reply:
This seems to be due to some documents not having entities in them (that happening once is enough to trigger the warning IIRC). So AFAIK this is not super relevant - when I checked most documents _had_ entities.

Reply:
Maybe we need to revisit when to raise this warning then. We currently raise it per batch on `update`, but maybe there's an earlier point in the pipeline when we could check whether there is *any* correct data, to avoid FP warnings like these.